File: org.kevoree.modeling.microframework/src/test/java/org/kevoree/modeling/microframework/test/cloud/impl/ElementImpl.java
package org.kevoree.modeling.microframework.test.cloud.impl;

import org.kevoree.modeling.api.abs.AbstractKObject;
import org.kevoree.modeling.api.data.KDataManager;
import org.kevoree.modeling.api.meta.MetaClass;
import org.kevoree.modeling.microframework.test.cloud.Element;
import org.kevoree.modeling.microframework.test.cloud.meta.MetaElement;

/**
 * Created by duke on 10/13/14.
 */
public class ElementImpl extends AbstractKObject implements Element {

    public ElementImpl(long p_universe, long p_time, long p_uuid, MetaClass p_metaClass, KDataManager p_manager) {
        super(p_universe, p_time, p_uuid, p_metaClass, p_manager);
    }

    @Override
    public String getName() {
        return (String) this.get(MetaElement.ATT_NAME);
    }

    @Override
    public Element setName(String p_name) {
        this.set(MetaElement.ATT_NAME, p_name);
        return this;
    }

    @Override
    public Double getValue() {
        return (Double) this.get(MetaElement.ATT_VALUE);
    }

    @Override
    public Element setValue(Double p_name) {
        this.set(MetaElement.ATT_VALUE, p_name);
        return this;
    }

}

File: org.kevoree.modeling.microframework.typescript/src/test/ts/org/kevoree/modeling/microframework/test/cloud/impl/ElementImpl.ts
///<reference path="../../../../api/KDimension.ts"/>
///<reference path="../../../../api/abs/AbstractKObject.ts"/>
///<reference path="../../../../api/meta/MetaAttribute.ts"/>
///<reference path="../../../../api/meta/MetaClass.ts"/>
///<reference path="../../../../api/meta/MetaOperation.ts"/>
///<reference path="../../../../api/meta/MetaReference.ts"/>
///<reference path="../../../../api/time/TimeTree.ts"/>
///<reference path="../CloudView.ts"/>
///<reference path="../Element.ts"/>
///<reference path="../Node.ts"/>

class ElementImpl extends AbstractKObject<Element, CloudView> implements Element {

  private _mataReferences: MetaReference[] = new Array();

  constructor(factory: CloudView, metaClass: MetaClass, kid: number, now: number, dimension: KDimension, timeTree: TimeTree) {
    super(factory, metaClass, kid, now, dimension, timeTree);
  }

  public metaAttributes(): MetaAttribute[] {
    return Element.METAATTRIBUTES.values();
  }

  public metaReferences(): MetaReference[] {
    return this._mataReferences;
  }

  public metaOperations(): MetaOperation[] {
    return new Array();
  }

  public getName(): string {
    return <string>get(METAATTRIBUTES.NAME);
  }

  public setName(name: string): Element {
    set(METAATTRIBUTES.NAME, name);
    return this;
  }

  public getValue(): number {
    return <number>get(METAATTRIBUTES.VALUE);
  }

  public setValue(name: number): Element {
    set(METAATTRIBUTES.VALUE, name);
    return this;
  }

}

